src/python/zquantum/core/interfaces/mock_objects.py

```diff
@@ -25,23 +29,25 @@ class MockQuantumBackend(QuantumBackend):
     def __init__(self, n_samples=None):
         super().__init__(n_samples)
 
-    def run_circuit_and_measure(self, circuit, n_samples=None, **kwargs):
+    @compatible_with_old_type(Circuit, new_circuit_from_old_circuit)
+    def run_circuit_and_measure(
+        self, circuit: NewCircuit, n_samples: int = None, **kwargs
```

Comment:
I think `n_samples` should be types as `Optional[int]`, because of its default `None` value.